### PR TITLE
Allow FlumpView URL mapping to be customised.

### PR DESCRIPTION
**Pull request by obmarg**
Why do we need this change?
----

It's not uncommon to need to write an API endpoint for which GET_MANY doesn't
make sense: for example, getting the current users profile data.  There's only
one current_user, so only one response makes sense.

Under these circumstances, it'd be nice to have the GET endpoint under `/`
rather than `/<entity_id>`.

What effects does this change have?
---

This adds a (currently undocumented) MAPPING dict to the FlumpView, which should
allow us to change the specific URL flump uses for each of the individual methods
on a FlumpView.

### CODE REVIEW

File: flump/__init__.py [hunk 1]
```diff
@@ -81,35 +81,21 @@ def register_flump_view(self, view_class, url):
         # on both anyway.
         url = url.rstrip('/')
 
-        get_many_func = (
-            view_func if HttpMethods.GET_MANY < methods else _meth_not_allowed
```
Comment (lukasz-madon) [inline flump/__init__.py]: I'm not sure what `<` mean. Is it short for `s.issubset(t)` and equivalent of ` s <= t`. Is it pyton3 syntax?

**Comment by lukasz-madon** [comment]
1. We should probably raise an issue to document that
2. Do we care about any REST principles e.g. using plural form.

**Comment by obmarg** [comment]
What do you mean "Do we care about rest principles"?

File: flump/__init__.py (hunk 1 above)
Comment (obmarg) [inline flump/__init__.py]: image

`<` was not quite short for issubset, `<=` is.

**Comment by obmarg** [comment]
Re: rest principles, we probably don't care much unless it's in the json API spec: http://jsonapi.org/

**Comment by lukasz-madon** [comment]
That's a bad question. Many REST API recommendations says that you should use plural form to describe resources e.g. /users will return a list with just a single item if there is just a single user (see second answer). This is a convention which is enforced in flump. I wonder if this change would break anything

File: flump/__init__.py (hunk 1 above)
Comment (lukasz-madon) [inline flump/__init__.py]: I can't see `<` in docs

**Comment by lukasz-madon** [comment]
LGTM

**Comment by obmarg** [comment]
I see.  Not sure that'd be sensible to enforce in flump:

- Json API explicitly says it's "agnostic about inflection rules", so I don't know if we should be forcing some rules apon anyone
- Not every endpoint is a collection.  This change has been made specifically to support endpoints that aren't collections, like the various sync timing endpoints in rp-app, a users settings etc.  Doesn't seem to make sense to force a "user_password" endpoint to be called "user_passwords" for example.
- Not every word pluralises by putting s on the end. (Most of them do, so this is a minor niggle, but a niggle all the same)

**Pull request merged**